### PR TITLE
Add enemy

### PR DESCRIPTION
**Pull request by Nicklason**

### CODE REVIEW

**Review by stped18 — APPROVED** [review]

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/LifePart.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        
+    }
+    public void addHitPoint(int amount){
+        this.hit_point+=amount;
+        if (this.hit_point>this.MAX_HIT_POINTS){
+            this.hit_point=this.MAX_HIT_POINTS;
+        }
+    }
+
+    public void setMAX_HIT_POINTS(int MAX_HIT_POINTS) {
```
Comment (Nicklason) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/LifePart.java:38]: Use camel case

**Review by DenBedste — APPROVED** [review]

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/MovingPart.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        }
+        if(DOWN){
+            p.getPoint().translate(0, -10);
+        }
+        if (RIGHT){
+            p.getPoint().translate(10, 0);
+        }
+        if(LEFT){
+            p.getPoint().translate(-10, 0);
+        }
```
Comment (Nicklason) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/MovingPart.java:58]: Moving diagonally allows you to move further than moving vertically or horizontally.

File: dk.sdu.se4.core/src/main/java/dk/sdu/se4/core/Game.java [hunk 3]
```diff
@@ -21,39 +43,106 @@ public Game() {
...
         this.mapService = mapService;
-        //System.out.println("ADDED MapService TO Game");
+
+        System.out.println("ADDED MapService TO Game to " + this.mapService.toString());
     }
 
     public void removeMapService(MapService mapService) {
         this.mapService = null;
-        //System.out.println("REMOVED MapService FROM Game");
+        System.out.println("REMOVED MapService FROM Game to " + this.mapService.toString());
```
Comment (Nicklason) [inline dk.sdu.se4.core/src/main/java/dk/sdu/se4/core/Game.java:115]: Should not be printing

File: dk.sdu.se4.core/src/main/java/dk/sdu/se4/core/Game.java [hunk 4]
```diff
@@ -21,39 +43,106 @@ public Game() {
...
-    public void dispose() {}
+    public void dispose() {
+
+    }
 
     public void addMapService(MapService mapService) {
         this.mapService = mapService;
-        //System.out.println("ADDED MapService TO Game");
+
+        System.out.println("ADDED MapService TO Game to " + this.mapService.toString());
```
Comment (Nicklason) [inline dk.sdu.se4.core/src/main/java/dk/sdu/se4/core/Game.java:110]: Should not be printing

File: dk.sdu.se4.enemy/src/main/java/dk/sdu/se4/enemy/EnemyCore.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ *
+ * @author steff
+ */
+public class EnemyCore {
+    protected MapService mapService=null;
+    
+    public void addMapService(MapService mapService) {
+        this.mapService = mapService;
+        System.out.println("ADDED MapService TO Enemy "+this.mapService.getClass().toString());
```
Comment (Nicklason) [inline dk.sdu.se4.enemy/src/main/java/dk/sdu/se4/enemy/EnemyCore.java:19]: Don't print

File: dk.sdu.se4.enemy/src/main/java/dk/sdu/se4/enemy/EnemyCore.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    public void removeMapService(MapService mapService) {
+        this.mapService = null;
+        System.out.println("REMOVED MapService FROM Enemy "+this.mapService.getClass().toString());
```
Comment (Nicklason) [inline dk.sdu.se4.enemy/src/main/java/dk/sdu/se4/enemy/EnemyCore.java:24]: Don't print

File: dk.sdu.se4.enemy/src/main/java/dk/sdu/se4/enemy/EnemyPlugin.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        System.out.println("loading Enemy");
+        if (this.mapService != null) {
+           for (int i =0; i<enemy.length;i++){
+            enemy[i] = new Enemy();
+            int x = (int) (Math.random() * 800) + 1;
+            int y = (int) (Math.random() * 600) + 1;
+            enemy[i].addPart(new PositionPart(new Point(x, y)));
+            enemy[i].addPart(new MovingPart());
+            enemy[i].addPart(new LifePart(100));
+            enemy[i].addPart(new ImagePart(new File("C:/Users/steff/OneDrive/Documents/GitHub/4-semester-project/dk.sdu.se4.enemy/src/main/resources/img/zombi.png"), 150, 150));
```
Comment (Nicklason) [inline dk.sdu.se4.enemy/src/main/java/dk/sdu/se4/enemy/EnemyPlugin.java:36]: Don't use absolute path

**Pull request merged**